Add genupi command

haoadoreorange opened the pull request:
Add shortcut `gui` command for generate - update - ignore in 1
Signed-off-by: HaotheAdorableOyster <github@comtam.dev>

Commit:
Add genupi command

Signed-off-by: HaotheAdorableOyster <github@comtam.dev>

haoadoreorange commented:
No idea why it failed (:

sp1thas commented on bin/dropboxignore.sh:
```diff
@@ -545,6 +545,7 @@ Usage: "$PROGRAM_NAME" <command> <path> [-v 0-2] [-p pattern]
 
   Commands:
 
+    gui			generate - update - ignore in 1 shortcut command
```
Well, `gui` abbreviation could be misunderstood because is inextricably linked with the concept of Graphical User Interface. I think that we should find a different action name. What do you think? My proposal is to adopt `combo` but feel free to propose a different name.

```suggestion
    combo               Generate, update and ignore at once.
```

sp1thas commented on bin/dropboxignore.sh:
```diff
@@ -644,15 +649,23 @@ main() {
   # check input file or folder
   check_input "$input_f"
-
+  raw_input_f=$input_f
```
Why we should use the raw input file/folder here? The new function seems to work fine with existing `$input_f`

sp1thas commented on bin/dropboxignore.sh:
```diff
@@ -644,15 +649,23 @@ main() {
-  if [ "$generate_action" == true ]; then
+  if [ "$gui_action" == true ]; then
+    gui_action=false
+    main generate $raw_input_f
```
Instead of using the low-level function, we could use the high-level function: `generate_dropboxignore_files`, `update_dropboxignore_files`, `ignore_files`

sp1thas reviewed (commented):
Thank you for this PR, this functionality looks very helpful. Regarding the failed github action, you should install [shellcheck](https://github.com/koalaman/shellcheck) in you OS and then check-out the formatting issues by running:

```shell
$ shellcheck bin/*
$ shellcheck tests/*
```

Finally, a new test case about this new action should be added. If you are not familiar with [bats](https://github.com/bats-core/bats-core) testing system please let me know :)

haoadoreorange commented:
Sorry for the delay, so I changed the name to `genupi` instead, and use the high-level function as suggested, also removed `raw_input`. Could you help with the test case please ?

sp1thas commented on bin/dropboxignore.sh:
```diff
@@ -545,6 +545,7 @@ Usage: "$PROGRAM_NAME" <command> <path> [-v 0-2] [-p pattern]
 
   Commands:
 
+    genupi		generate - update - ignore in 1 shortcut command
```
```suggestion
    genupi		  Generate, update & ignore using one shortcut command.
```

sp1thas commented:
Regarding the test, add the following test in [tests/tests.bats:L95](https://github.com/sp1thas/dropboxignore/blob/master/tests/tests.bats#L95)

```shell
@test "Test genupi command" {
  run $dropboxignore genupi "$TEST_FOLDER"
  assert_success
  assert_file_exist "$TEST_FOLDER/$DROPBOXIGNORE_NAME"
  assert_file_exist "$TEST_FOLDER/$GITIGNORE_NAME"
  assert_output --partial "Total number of generated files: 0"
}
```

Commit:
Update help message

Co-authored-by: Panagiotis Simakis <sp1thas@autistici.org>

Commit:
Add test case for genupi

Signed-off-by: HaotheAdorableOyster <github@comtam.dev>

haoadoreorange commented:
> ```shell
> @test "Test genupi command" {
>   run $dropboxignore genupi "$TEST_FOLDER"
>   assert_success
>   assert_file_exist "$TEST_FOLDER/$DROPBOXIGNORE_NAME"
>   assert_file_exist "$TEST_FOLDER/$GITIGNORE_NAME"
>   assert_output --partial "Total number of generated files: 0"
> }
> ```

done !
